### ISSUE TITLE
Make critical jobs Guaranteed Pod QOS: pull-kubernetes-bazel-test

### ISSUE THREAD

**Issue by spiffxp** [kubernetes/test-infra#18584]
**What should be cleaned up or changed**:

This is part of https://github.com/kubernetes/test-infra/issues/18530

The following jobs should be Guaranteed Pod QOS, meaning they should have CPU and memory resource limits, and matching resource requests:
- pull-kubernetes-bazel-test

These jobs run on (google.com only) k8s-prow-build, so @spiffxp has provided the following guess:
- suggest 4 cpu, slighly above 36 Gi (38?) mem

General steps to follow:
- update the job definitions in [/config/jobs](/config/jobs) to have matching CPU and memory limits and requests
  - e.g. ([ci-kubernetes-e2e-gci-gce has a `resources:` field with matching entries](https://github.com/kubernetes/test-infra/blob/2eac54f721ed58479a3126e4f9ca5cbfcc73821b/config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml#L357-L394))
- open a pull request, include a link to this issue and cc `@kubernetes/ci-signal` in the description
- keep an eye on the jobs for the next few days, and if they start failing more than usual, open followup pull requests to raise resources
  - can look at recent prow.k8s.io runs (e.g. https://prow.k8s.io/?job=ci-kubernetes-e2e-gci-gce)
  - can look at job history (e.g. https://prow.k8s.io/job-history/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce)
  - can look at testgrid dashboards (e.g. https://testgrid.k8s.io/sig-release-master-blocking#gce-cos-master-default)
- after the job has remained stable for a few days, declare victory

/sig testing
/sig release
/area jobs
/area release-eng

**Comment by RobertKielty** [comment]
Hi @spiffxp 
Hi @hasheddan 

https://prow.k8s.io/?job=pull-kubernetes-bazel-test
https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-bazel-test


I'm looking at the duration graph on this job ... 
https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-bazel-test&graph-metrics=test-duration-minutes&width=5

It seems that we are getting more duration spikes in excess of 30 min  over the last 4 to 8 days. 

Can one of you take a look at this?

**Comment by spiffxp** [comment]
So for what it's worth, this job currently uses [RBE](https://blog.bazel.build/2018/10/05/remote-build-execution.html) which means the majority of test execution is happening elsewhere.  RBE is alpha only and not currently available over in k8s infra.  We're going to have to switch away from RBE to migrate the job over there.  This is what I've done for periodic-kubernetes-bazel-test (ref: https://github.com/kubernetes/test-infra/issues/18652), and [its average runtime jumped up from 10min to 30min](https://github.com/kubernetes/test-infra/issues/18652#issuecomment-670970119)

tl;dr we should eventually expect the job to take over 30min, so I'm wary of spending too much time to get it back down to 10min here

But, I agree this is unexpected behavior.  I have not really being able to catch resource contention happening, but will see if anything pops up for this job

**Comment by RobertKielty** [comment]
Just checked this again, job duration's have dropped back down to ranging from 10 to 20 mins

**Comment by spiffxp** [comment]
/close
Sounds good.  I'm not going to have time to dig deeper into this, I think we can call this good enough for now.
